Description: Add missing alias definitions for VMIN and VTIME on sparc64
 On sparc and sparc64, VMIN and VTIME in termios are defined as
 aliases of VEOF and VTIME respectively. Thus, in order to be able
 to use VMIN and VTIME in other packages like rustyline, we need to
 add this alias definitions in nix as well. Since enums cannot have
 the same numerical value defined twice, we use associated constants
 to define the aliases for VEOF and VEOL in SpecialCharacterIndices.
 .
Author: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>

---

Forwarded: https://github.com/nix-rust/nix/pull/1151
Forwarded: https://github.com/nix-rust/nix/pull/1150
Last-Update: 2019-11-01

Index: rust-nix-0.15.0/src/sys/termios.rs
===================================================================
--- rust-nix-0.15.0.orig/src/sys/termios.rs
+++ rust-nix-0.15.0/src/sys/termios.rs
@@ -583,6 +583,13 @@ libc_enum! {
     }
 }
 
+impl SpecialCharacterIndices {
+    #[cfg(all(target_os = "linux", target_arch = "sparc64"))]
+    pub const VMIN: SpecialCharacterIndices = SpecialCharacterIndices::VEOF;
+    #[cfg(all(target_os = "linux", target_arch = "sparc64"))]
+    pub const VTIME: SpecialCharacterIndices = SpecialCharacterIndices::VEOL;
+}
+
 pub use libc::NCCS;
 #[cfg(any(target_os = "dragonfly",
           target_os = "freebsd",
